### PR TITLE
[go-gen] Refactor go code generation by package

### PR DESCRIPTION
**Pull request by willjvsmith**
* Replaces `Seqs` with `mkCode.lines` in the `generateMain` function in `GoServiceMainGenerator`
* Refactors the code generation by moving each go package to be generated into their own files

No new code generation here

### CODE REVIEW

**Review by jaylees14 — APPROVED** [review]

File: src/main/scala/temple/generate/service/go/GoServiceMainGenerator.scala [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    mkCode.lines(
+      s"var dao ${serviceName}DAO.DAO",
+      when(usesComms) { s"var comm ${serviceName}Comm.Handler" },
+    )
+
+  /** Given a service name, whether the service uses inter-service communication, the endpoints desired and the port
+    * number, generate the main function */
+  private[go] def generateMain(serviceName: String, usesComms: Boolean, endpoints: Set[Endpoint], port: Int): String = {
+    val sb = new StringBuilder
+    sb.append("func main() ")
```
Comment (c-harding) [inline src/main/scala/temple/generate/service/go/GoServiceMainGenerator.scala]: This is unused, right?

**Review by c-harding — COMMENTED** [review]
Nice, just one redundant variable

File: src/main/scala/temple/generate/service/go/GoServiceMainGenerator.scala [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
```
Comment (c-harding) [inline src/main/scala/temple/generate/service/go/GoServiceMainGenerator.scala]:
```suggestion
```

**Review by Lewiky — APPROVED** [review]
OOOH YEAH. I LIKE THAT<><><<><>>>

File: src/main/scala/temple/generate/service/go/GoServiceMainGenerator.scala [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+      "valid.SetFieldsRequiredByDefault(true)",
+      "",
+      "config, err := util.GetConfig(*configPtr)",
+      "if err != nil " + CodeWrap.curly.tabbed("log.Fatal(err)"),
+      "",
+      s"dao = ${serviceName}DAO.DAO{}",
+      "err = dao.Init(config)",
+      "if err != nil " + CodeWrap.curly.tabbed("log.Fatal(err)"),
+      "",
+      when[CodeTerm](usesComms) {
```
Comment (c-harding) [inline src/main/scala/temple/generate/service/go/GoServiceMainGenerator.scala]:
```suggestion
      when(usesComms) {
```

File: src/main/scala/temple/generate/service/go/GoServiceMainGenerator.scala (hunk 1 above)
Comment (willjvsmith) [reply src/main/scala/temple/generate/service/go/GoServiceMainGenerator.scala]: Yeah this should have been removed in your refactor PR, I'll remove it here.

Though in general I've tried to keep the order of code generating the same as the order of code, i.e. `"fun main() "` and then the body

Comment (willjvsmith) [reply src/main/scala/temple/generate/service/go/GoServiceMainGenerator.scala]: I guess we don't need the `body` variable, and we can maintain the order

File: src/main/scala/temple/generate/service/go/GoServiceMainGenerator.scala [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        mkCode.lines(
+          s"comm = ${serviceName}Comm.Handler{}",
+          "comm.Init(config)",
+          "",
+        )
+      },
+      "r := mux.NewRouter()",
+      when[CodeTerm](endpoints.contains(Endpoint.ReadAll)) {
```
Comment (c-harding) [inline src/main/scala/temple/generate/service/go/GoServiceMainGenerator.scala]:
```suggestion
      when(endpoints.contains(Endpoint.ReadAll)) {
```

**Review by c-harding — COMMENTED** [review]
The type parameters also aren’t needed here (as the contents are strings, and so don’t need to be cast)

File: src/main/scala/temple/generate/service/go/GoServiceMainGenerator.scala (hunk 1 above)
Comment (c-harding) [reply src/main/scala/temple/generate/service/go/GoServiceMainGenerator.scala]: Yeah, this was my fault, sorry

**Review by c-harding — APPROVED** [review]

**Pull request merged**